File: sdk-android/src/main/java/cu/suitetecsa/sdk/android/balance/consult/UssdRequest.java
package cu.suitetecsa.sdk.android.balance.consult;

import android.net.Uri;

public enum UssdRequest {
    PRINCIPAL_BALANCE("*222" + Uri.parse("#")),
    DATA_BALANCE("*222*328" + Uri.parse("#")),
    VOICE_BALANCE("*222*869" + Uri.parse("#")),
    MESSAGES_BALANCE("*222*767" + Uri.parse("#")),
    BONUS_BALANCE("*222*266" + Uri.parse("#")),
    CUSTOM("Nothing");

    private final String ussdCode;

    UssdRequest(String ussdCode) {
        this.ussdCode = ussdCode;
    }

    public String getUssdCode() {
        return ussdCode;
    }
}

File: sdk-android/src/main/java/io/github/suitetecsa/sdk/android/balance/consult/UssdRequest.kt
package io.github.suitetecsa.sdk.android.balance.consult

import androidx.core.net.toUri

enum class UssdRequest(val ussdCode: String) {
    BALANCE("*222" + "#".toUri()),
    DATA("*222*328" + "#".toUri()),
    VOICE("*222*869" + "#".toUri()),
    SMS("*222*767" + "#".toUri()),
    BONUSES("*222*266" + "#".toUri()),
    CUSTOM("Nothing")
}
